src/string.cr

```diff
@@ -3586,21 +3598,43 @@ class String
     match_result.try &.begin
   end
 
-  # :ditto:
-  #
+  # Returns the index of the _last_ appearance of *search* in the string,
+  # If *offset* is present, it defines the position to _end_ the search
+  # (characters beyond this point are ignored).
   # Raises `Enumerable::NotFoundError` if *search* does not occur in `self`.
-  def rindex!(search : Regex, offset = size, *, options : Regex::MatchOptions = Regex::MatchOptions::None) : Int32
-    rindex(search, offset, options: options) || raise Enumerable::NotFoundError.new
+  #
+  # ```
+  # "Hello, World".rindex('o') # => 8
+  # "Hello, World".rindex('Z') # => Enumerable::NotFoundError
```

Comment:
**suggestion:** Proper method name. And conventional syntax for raised exceptions.
```suggestion
  # "Hello, World".rindex!('o') # => 8
  # "Hello, World".rindex!('Z') # raises Enumerable::NotFoundError
```

Ditto below.